core/field_colour_slider.js

```diff
@@ -45,343 +35,364 @@ goog.require('goog.ui.Slider');
+    node.style['background'] = `linear-gradient(to right, ${gradient})`;
   }
-  this.updateSliderHandles_();
-  this.updateDom_();
-};
 
-/**
- * Create the hue, saturation or value CSS gradient for the slide backgrounds.
- * @param {string} channel – Either "hue", "saturation" or "value".
- * @return {string} Array colour hex colour stops for the given channel
- * @private
- */
-Blockly.FieldColourSlider.prototype.createColourStops_ = function(channel) {
-  var stops = [];
-  for(var n = 0; n <= 360; n += 20) {
-    switch (channel) {
-      case 'hue':
-        stops.push(goog.color.hsvToHex(n, this.saturation_, this.brightness_));
-        break;
-      case 'saturation':
-        stops.push(goog.color.hsvToHex(this.hue_, n / 360, this.brightness_));
-        break;
-      case 'brightness':
-        stops.push(goog.color.hsvToHex(this.hue_, this.saturation_, 255 * n / 360));
-        break;
-      default:
-        throw new Error("Unknown channel for colour sliders: " + channel);
+  /**
+   * Update the readouts and slider backgrounds after value has changed.
+   * @private
+   */
+  updateDom_() {
+
+    if (this.hueSlider_) {
+      // Update the slider backgrounds
+      this.setGradient_(this.hueSlider_, 'hue');
+      this.setGradient_(this.saturationSlider_, 'saturation');
+      this.setGradient_(this.brightnessSlider_, 'brightness');
+
+      // Update the readouts
+      this.hueReadout_.textContent = Math.floor(100 * this.hue_ / 360).toFixed(0);
+      this.saturationReadout_.textContent = Math.floor(100 * this.saturation_).toFixed(0);
+      this.brightnessReadout_.textContent = Math.floor(100 * this.brightness_ / 255).toFixed(0);
     }
   }
-  return stops;
-};
 
-/**
- * Set the gradient CSS properties for the given node and channel
- * @param {Node} node - The DOM node the gradient will be set on.
- * @param {string} channel – Either "hue", "saturation" or "value".
- * @private
- */
-Blockly.FieldColourSlider.prototype.setGradient_ = function(node, channel) {
-  var gradient = this.createColourStops_(channel).join(',');
-  goog.style.setStyle(node, 'background',
-      '-moz-linear-gradient(left, ' + gradient + ')');
-  goog.style.setStyle(node, 'background',
-      '-webkit-linear-gradient(left, ' + gradient + ')');
-  goog.style.setStyle(node, 'background',
-      '-o-linear-gradient(left, ' + gradient + ')');
-  goog.style.setStyle(node, 'background',
-      '-ms-linear-gradient(left, ' + gradient + ')');
-  goog.style.setStyle(node, 'background',
-      'linear-gradient(left, ' + gradient + ')');
-};
+  /**
+   * Update the slider handle positions from the current field value.
+   * @private
+   */
+  updateSliderHandles_() {
+    if (this.hueSlider_) {
+      // Don't let the following calls to setValue for each of the sliders
+      // trigger the slider callbacks (which then call setValue on this field again
+      // unnecessarily)
+      this.sliderCallbacksEnabled_ = false;
```

Comment:
I don't think setting values programatically triggers input events. So can you try getting rid of this now that we're not in closure land?

Reply:
Done!